### PR TITLE
Jakarta Data 1.0

### PR DESCRIPTION
**Pull request by KyleAure**
## Specification PR template
When creating a specification project release review, create PRs with the content defined as follows.

Include the following in the PR:
- [x] A directory in the form wombat/x.y where x.y is the release major.minor version, and the directory contains the following.
- [x] Specification PDF in the form of jakarta-wombat-spec-x.y.pdf
- [x] Specification HTML in the form of jakarta-wombat-spec-x.y.html
- [x] A specification page named _index.md following the template at:
      https://github.com/jakartaee/specification-committee/blob/master/spec_page_template.md
- [x] For a Progress Review, that sufficient progress has been made on a Compatible Implementation and TCK, to ensure that the spec is implementable and testable.
- [x] For a [Release Review](https://www.eclipse.org/projects/handbook/#release-review), a summary that a Compatible Implementation is complete, passes the TCK, and that the TCK includes sufficient coverage of the specification. The TCK users guide MUST include the instructions to run the compatible implementations used to validate the release.
Instructions MAY be by reference.
   - [x] Updated release record - https://jakarta.ee/specifications/data/1.0/
   - [x] ~Generated IP Log~
   - [x] Email to PMC - https://projects.eclipse.org/projects/ee4j.data/reviews/1.0-release-review
   - [ ] Start release review by emailing EMO 
- [x] The URL of the OSSRH staging repository for the api, javadoc:
      - https://jakarta.oss.sonatype.org/content/repositories/jakartadata-1023/
- [x] The URL of the staging directory on downloads.eclipse.org for the proposed EFTL TCK binary:
      - Staged: https://download.eclipse.org/ee4j/data/jakartaee11/staged/eftl/
      - Promoted: https://download.eclipse.org/ee4j/data/jakartaee11/promoted/eftl/
- [x] The URL of the compatibility certification request issue:
      - Persistence (standalone): https://github.com/jakartaee/data/issues/714
      - Persistence (platform)    : https://github.com/jakartaee/data/issues/741
- [x] Specification JavaDoc in the wombat/x.y/apidocs directory. 
If desired, an optional second PR can be created to contain just the JavaDoc in the `apidocs` directory.

Note: If any item does not apply, check it and mark N/A below it.

### CODE REVIEW

**Comment by mtdelgadoa** [comment]
Hello all, 
the EMO is trying to avoid duplication of release records, so we will be providing our feedback and comments related to this release directly here.

## EMO REVIEW CHECKLIST 

EDP Review status: ONGOING

### **EMO review checklist**
PMI record: https://projects.eclipse.org/projects/ee4j.data/reviews/1.0-release-review

_**EF Specification Process**_      
- [ ] Spec Committee Ballot completed

_**Intellectual Property Management**_      
- [ ] All project code has [copyright and license headers](https://www.eclipse.org/projects/handbook/#ip-copyright-headers) correctly applied. ** EMO will scan the code at their discretion **   
- [ ]  All distributed third-party content has been vetted by the IP Due Diligence process (i.e., IP Log has been approved)   

_**Open Source Rules of Engagement**_
- [x] PMC approval [on the mailing list](https://github.com/eclipsefdn/emo-team/emo/-/blob/main/%7Blink%20to%20the%20message%7D)

General:
- [x] Project is operating within the mission and scope defined in its top-level project’s charter   
- [x]  Project is operating within the bounds of its own scope   
- [x]  Project is operating in an open and transparent manner       
- [x]  Overall the project is operating according to the Eclipse Development Process.   

Things to check:
- Communication channels advertised
- Advertised communication channels used
- Committers are responding to questions
- Committers are responding to issues
- Committers are responding to pull/merge/review requests

_**Branding and Trademarks**_
The following applies when the project has a custom website.
To the best of our knowledge:
- [x]  Project content correctly uses Eclipse Foundation trademarks    
- [x] Project content (code and documentation) does not violate trademarks owned by other organizations   

Things to check:
- [ ]  Project website uses the [project's formal name](https://www.eclipse.org/projects/handbook/#trademarks-website-name) in first and all prominent references
- [ ]   Project website includes a trademark attribution statement
- [ ] Project website footers contain all [necessary elements](https://www.eclipse.org/projects/handbook/#trademarks-website-footer)

_**Legal Documentation**_
Required files:   
- [ ] License files in all repository roots   
- [ ]  README       
- [ ]  CONTRIBUTING (or equivalent)   

Recommended files:
- [ ] NOTICES or equivalent (you can use the [Legal Documentation generator](https://www.eclipse.org/projects/tools/documentation.php) available under committer tools)   
- [ ]  CODE_OF_CONDUCT   
- [ ]  SECURITY   
 
See examples for [Security file](https://gitlab.eclipse.org/security/best-practices/-/blob/main/templates/SECURITY.md) and [Code of Conduct](https://github.com/eclipse/.github/blob/master/CODE_OF_CONDUCT.md).

Required elements:
- [ ]  ECA is referenced/described   

Recommended elements:
- [ ]  Build instructions provided   
- [ ] [Security policy](https://www.eclipse.org/security/policy.php) is described   
 
_**Metadata (PMI)**_
- [ ] The formal name, e.g. "Eclipse Foo™", is used in the project title   
- [ ] The formal name including appropriate marks (e.g, "™") is used in the first mention in the text of the project description, and scope   
- [ ] The project description starts with a single paragraph that can serve as an executive summary   
- [ ] Source code repository references are up-to-date   
- [ ] Download links and information are up-to-date (see [EF handbook for more information on how-to do this](https://www.eclipse.org/projects/handbook/#resources-downloads))   
- [ ] Communication channels listed in the PMI (i.e. public mailing list, forums, etc.)

**Comment by Emily-Jiang** [comment]
### Mentor Spec Review Checklist

1. Spec PR
  - [ ] PR uses [template](https://github.com/jakartaee/specifications/blob/master/pull_request_template.md)
  - [ ] Directory of form {spec}/x.y
  - [ ] PDF of form jakarta-{spec}-spec-x.y.pdf ("-spec" preferred but not required)
  - [ ] HTML of form jakarta-{spec}-spec-x.y.html ("-spec" preferred but not required)
  - [ ] Index page {spec}/x.y/_index.md following [template](https://github.com/jakartaee/specification-committee/blob/master/spec_page_template.md)
  - [ ] Index page {spec}/_index.md following [template](https://github.com/jakartaee/specification-committee/blob/master/spec_index_template.md)
  - [ ] No other files (e.g., no jakarta_ee_logo_schooner_color_stacked_default.png)
  - [ ] Staging repository link of the form https://jakarta.oss.sonatype.org/content/repositories/staging/jakarta/{spec}/jakarta.{spec}-api/x.y.z/
  - [ ] EFTL TCK link of the form http://download.eclipse.org/.../+.zip
  - [ ] Compatibility certification link of the form https://github.com/eclipse-ee4j/{project}/#{issue}
  - [ ] N/A (Optional) Second PR for just apidocs

2. _index.md
  - [ ] Link to project release plan of the form https://projects.eclipse.org/projects/ee4j.{spec}/releases/x.y/plan
  - [ ] Link to spec pdf
  - [ ] Link to spec html
  - [ ] Link to apidocs
  - [ ] Link to final TCK download zip file of the form https://download.eclipse.org/jakartaee/{spec}/x.y/*{spec}-tck-x.y.z.zip (The folder path is required, the file name pattern is preferred.)
  - [ ] Link to API jar file of the form https://search.maven.org/artifact/jakarta.{spec}/jakarta.{spec}-api/x.y.z/jar
  - [ ] Name of and link to at least one Compatible Implementation
  - [ ] Review summary statement to ensure it has been updated to reflect the work accomplished and does not contain proposal statements as might be found in Plan Review statement.

3. javadocs
  - [ ] Footer contains Eclipse copyright and link to license
  - [ ] ESFL license is included, usually as doc-files/speclicense.html
  - [ ] no META-INF directory in PR
  - [ ] javadocs-jar artifact matches apidocs (optional for this release)

4. Spec PDF
  - [ ] Correct spec title
  - [ ] Version number of the form x.y, not x.y.z
  - [ ] Correct Eclipse copyright line
  - [ ] No DRAFT or SNAPSHOT
  - [ ] Correct Logo

5. Spec HTML
  - [ ] Same as PDF

6. TCK zip file
  - [ ] README file (optional for this release)
  - [ ] EFTL license file, preferably named LICENSE.md
  - [ ] User's Guide (or equivalent documentation)
  - [ ] How to test the Compatible Implementation(s) listed in _index.md above with the TCK (may be in UG)

7. TCK User's Guide (or equivalent documentation)
  - [ ] Software requirements listed
  - [ ] Installation and configuration described
  - [ ] How to run tests
  - [ ] Where to file challenges

8. Compatibility certification request
  - [ ] Request follows [template](https://github.com/jakartaee/specification-committee/blob/master/compatibility-certification-request.md)
  - [ ] SHA-256 fingerprint matches staged TCK zip file
  - [ ] Request issue has `certification` label.

9. TCK results summary
  - [ ] Page is hosted by Compatible Implementation project
  - [ ] Includes all information from certification request
  - [ ] Summary includes number of tests passed, failed, errors
  - [ ] SHA-256 fingerprint matches staged TCK zip file on cert request

10. If a Release Review is required, the specification project team contacts the EMO to initiate the review by sending an email to emo@eclipse.org.
    (A Release Review is not required if the current release is a Service Release based on a previously successful Major or Minor
    release as indicated by a release record on the project's Releases page, e.g., the [Jakarta Servlet releases page](https://projects.eclipse.org/projects/ee4j.servlet/reviews).)
    - [ ] An issue will be created by the EMO to track the release review.
    - [ ] The specification project team requests approval for the release from the PMC by sending an email to ee4j-pmc@eclipse.org.
    - [ ] The specification project team then delivers an IP Log to the IP Team for their review as described in https://www.eclipse.org/projects/handbook/#pmi-commands-iplog.

11. Update Jakarta EE API jar
  - [ ] Update the Jakarta EE API jar by submitting a PR to the jakartaee-api project that updates the version number of your API jar file.

**Comment by Emily-Jiang** [comment]
Mentor review comments:

- PDF of form jakarta-{spec}-spec-x.y.pdf

The addendum doc uses the name jakarta-method-name-query-1.0.pdf. Please update to jakarta-data-method-name-query-1.0.pdf and .html

File: data/1.0/_index.md
Comment (Emily-Jiang) [inline data/1.0/_index.md:42]: This is very confusion and they don't match with the formal term. I would rather delete them.

File: data/1.0/_index.md
Comment (Emily-Jiang) [inline data/1.0/_index.md:43]: I am not sure what `any` means. If you have `any`, why do you need `persistence` and `nosql`. I would state: `persistence`, `nosql`.

File: data/1.0/_index.md
Comment (Emily-Jiang) [inline data/1.0/_index.md:57]:
```suggestion
  * [Jakarta Data 1.0 Specification Addendum](./jakarta-data-addendum-1.0.pdf) (PDF)
```
Why don't you merge this in the overall jakarta-data-1.0.pdf? Having multiple docs is confusing. Is this doc optional?

File: data/1.0/_index.md
Comment (Emily-Jiang) [inline data/1.0/_index.md:66]: need to add the ratification link

File: data/1.0/_index.md
Comment (Emily-Jiang) [inline data/1.0/_index.md:67]: TBD needs to be updated.

File: data/1.0/_index.md
Comment (Emily-Jiang) [inline data/1.0/_index.md:71]: Please add a link to the compatible implementation

File: data/1.0/_index.md
Comment (KyleAure) [reply data/1.0/_index.md:57]: The `method name query` addendum is separate because it is part of the specification we intend to deprecate in the future but is included in the 1.0 version of the specification to offer a migration path for Spring users.

**Comment by Emily-Jiang** [comment]
@KyleAure The TCK license needs to use EFTCK v1.1. The file should be License.md not license.txt. You can find the license [here](https://www.eclipse.org/legal/tck.php).

**Comment by KyleAure** [comment]
@Emily-Jiang I've opened a PR against data to fix the TCK-Distribution license to version 1.1 here: https://github.com/jakartaee/data/pull/745

The link you provided to the license document has a copy of the license in HTML and AsciiDoc.  I replaced license.txt with license.adoc which should be consistent with the requirements. 
@edbratt - is currently in the process of asking if we need to make this update now or if it can wait for a service/major release because the same issue exists in Concurrency.

**Comment by KyleAure** [comment]
@Emily-Jiang I have created an issue to do this: https://github.com/jakartaee/data/issues/746
This requires more than a documentation update since the documentation is bound to actual code in the TCK that uses the colloquialism `full profile`

**Comment by Emily-Jiang** [comment]
> @Emily-Jiang I've opened a PR against data to fix the TCK-Distribution license to version 1.1 here: [jakartaee/data#745](https://github.com/jakartaee/data/pull/745)
> 
> The link you provided to the license document has a copy of the license in HTML and AsciiDoc versions. I replaced license.txt with license.adoc which should be consistent with the requirements. @edbratt - is currently in the process of asking if we need to make this update now or if it can wait for a service/major release because the same issue exists in Concurrency.

We discussed on the Jakarta EE spec committee call. From now on, all new spec releases need to use the v1.1.

**Comment by KyleAure** [comment]
> From now on, all new spec releases need to use the v1.1.
@Emily-Jiang  From now on... does that mean EE 11 or EE 12?

**Comment by Emily-Jiang** [comment]
> @Emily-Jiang I've opened a PR against data to fix the TCK-Distribution license to version 1.1 here: [jakartaee/data#745](https://github.com/jakartaee/data/pull/745)
> 
> The link you provided to the license document has a copy of the license in HTML and AsciiDoc versions. I replaced license.txt with license.adoc which should be consistent with the requirements. @edbratt - is currently in the process of asking if we need to make this update now or if it can wait for a service/major release because the same issue exists in Concurrency.

I double checked the minutes from Jakarta EE Spec Committee meetings. [Here](https://github.com/jakartaee/specification-committee/issues/79) is the latest statement regarding the license version requirement. Using TCK License v1.1 is encouraged but not mandatory. Since updating the tck jar causes CCRs to be redone, I am happy for this project to use v1.0 for this release. Any future releases must update the license to v1.1.

**Comment by Emily-Jiang** [comment]
> @Emily-Jiang I have created an issue to do this: [jakartaee/data#746](https://github.com/jakartaee/data/issues/746) This requires more than a documentation update since the documentation is bound to actual code in the TCK that uses the colloquialism `full profile`

It is okay to have it corrected in the next release as you cannot easily release another TCK jar without a big delay.

**Review by Emily-Jiang — APPROVED** [review]

**Comment by Emily-Jiang** [comment]
On ballot completion, the specification committee mentor:

 - [x] adds this final checklist to the main PR.
 - [x] adds the `approved` label to the PRs, and sends out the Ballot Summary per this [template](#ballot-summary-email-template) to the [public Jakarta EE Specification Committee email list](jakarta.ee-spec@eclipse.org)
 - [ ] calculates the staged EFTL TCK signature and promotes it to the committee download area
  using the https://ci.eclipse.org/jakartaee-spec-committee/job/promote-release/ job. Manually editing the jenkins Build Information will help identify the build (ie. Mail 2.0 or CDI 3.0).
 - [x] merges the specification (and apidocs) PRs, ensuring the "date:" field in the _index.md file has an appropriate value to allow publishing.
 - [ ] updates the specification page with the ballot results. This is normally done via a separate PR that should be reviewed, approved, and merged.
 - [x] notifies the EMO of the ballot results by email to [emo@eclipse-foundation.org](emo@eclipse-foundation.org). Just forward the ballot summary note sent earlier to the public Spec Committee email list.
 - [ ] creates an issue in the specification project that includes the following checklist for the specification project team:

**Pull request merged**
